examples/ui/text_cursor.rs

```diff
@@ -0,0 +1,109 @@
+            height: Val::Percent(100.),
+            justify_content: JustifyContent::Center,
+            align_items: AlignItems::Center,
+            row_gap: Val::Px(50.),
+            flex_direction: FlexDirection::Column,
+            ..Default::default()
+        })
+        .with_child((
+            // Accepts a `String` or any type that converts into a `String`, such as `&str`
+            Text::new(
+                "Lorem ipsum dolor sit amet,\nconsectetur adipiscing elit,\nsed do eiusmod tempor incididunt\nut labore et dolore magna aliqua."
+            ),
+            TextFont {
+                // This font is loaded and will be used instead of the default font.
+                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                font_size: 30.0,
+                ..default()
+            },
+            TextColor(GOLDENROD.into()),
+            TextCursor {
+                line: 0,
+                index: 0,
+            },
+            // Set the justification of the Text
+            TextLayout::new_with_justify(JustifyText::Center),
+            // Set the style of the Node itself.
+            Outline {
+                color: Color::WHITE,
+                width: Val::Px(2.),
+                offset: Val::Px(25.),
+            },
+        ))
+        .with_child((
+            // Accepts a `String` or any type that converts into a `String`, such as `&str`
+            Text::new(
+                "Lorem ipsum dolor sit amet,\nconsectetur adipiscing elit,\nsed do eiusmod tempor incididunt\nut labore et dolore magna aliqua."),
+            TextFont {
+                // This font is loaded and will be used instead of the default font.
+                font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                font_size: 30.0,
+                ..default()
+            },
+            TextCursor {
+                line: 0,
+                index: 0,
+            },            
+            TextColor(GOLDENROD.into()),
+            // Set the justification of the Text
+            TextLayout::new_with_justify(JustifyText::Center),
+            // Set the style of the Node itself.
+            Outline {
+                color: Color::WHITE,
+                width: Val::Px(2.),
+                ..Default::default()
+            },
+        ))
+        ;
+}
+
+fn move_cursor(
+    keys: Res<ButtonInput<KeyCode>>,
+    mut cursors: Query<(&mut TextCursor, &mut TextCursorStyle)>,
+) {
+    for (mut cursor, mut style) in &mut cursors {
+        if keys.just_pressed(KeyCode::ArrowLeft) {
+            cursor.index = cursor.index.saturating_sub(1);
+        }
+        if keys.just_pressed(KeyCode::ArrowRight) {
+            cursor.index += 1;
+        }
+        if keys.just_pressed(KeyCode::KeyQ) {
+            cursor.line = cursor.line.saturating_sub(1);
+        }
+        if keys.just_pressed(KeyCode::ArrowDown) {
+            cursor.line += 1;
+        }
+        if keys.just_pressed(KeyCode::Insert) {
```

Comment:
Can we use a key that you can find on a laptop keyboard? And add some instructions?